src/main/java/core/basesyntax/strategy/DataConverterImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Action;
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String SEPARATION_PLACE = ",";
+    private static final int ACTION_PART = 0;
+    private static final int FRUIT_PART = 0;
+    private static final int QUANTITY_PART = 0;
+
+    public List<FruitTransaction> convertToTransactions(List<String> dateList) {
+        if (dateList == null) {
```

Comment:
The exception message 'Empty data list' might be misleading because the list isn't empty but rather null. A more accurate message would be 'Data list cannot be null'.